lib/compute/region.js

```diff
@@ -215,6 +221,198 @@ Region.prototype.createAddress = function(name, options, callback) {
+      var subnetworkInstance = self.subnetwork(subnetwork.name);
+      subnetworkInstance.metadata = subnetwork;
+      return subnetworkInstance;
+    });
+
+    callback(null, subnetworks, nextQuery, resp);
+  });
+};
+
+/**
+ * Get a reference to a Google Compute Engine address in this region.
+ *
+ * @resource [Instances and Networks]{@link https://cloud.google.com/compute/docs/instances-and-network}
+ *
+ * @param {string} name - Name of the subnetwork.
+ * @return {module:compute/subnetwork}
+ *
+ * @example
+ * var subnetwork = region.subnetwork('subnetwork-name');
+ */
+Region.prototype.subnetwork = function(name) {
+  return new Subnetwork(this, name);
+};
+
+/**
+ * Create a subnetwork in this region.
+ *
+ * @resource [Subnetwork Resource]{@link https://cloud.google.com/compute/docs/reference/v1/subnetworks#resource}
+ * @resource [Subnetwork: insert API Documentation]{@link https://cloud.google.com/compute/docs/reference/v1/subnetworks/insert}
+ *
+ * @param {string} name - Name of the subnetwork.
+ * @param {object} config - See a
+ *     [Subnetwork resource](https://cloud.google.com/compute/docs/reference/v1/subnetworks#resource).
+ * @param {string=} config.description - An optional description of this
+ *     resource. Provide this property when you create the resource.
+ * @param {string} config.ipCidrRange - The range of internal addresses that
+ *    are owned by this subnetwork. Provide this property when you create the
+ *    subnetwork. For example, 10.0.0.0/8 or 192.168.0.0/16. Ranges must be
+ *    unique and non-overlapping within a network.
+ * @param {string} config.network - The URL of the network to which this
+ *    subnetwork belongs, provided by the client when initially creating
+ *    the subnetwork. Only networks that are in the distributed mode can have
+ *    subnetworks.
+ * @param {string=} config.region - URL of the region where the Subnetwork
+ *    resides.
+ * @param {function} callback - The callback function.
+ * @param {?error} callback.err - An error returned while making this request.
+ * @param {module:compute/rule} callback.rule - The created Rule object.
+ * @param {module:compute/operation} callback.operation - An operation object
+ *     that can be used to check the status of the request.
+ * @param {object} callback.apiResponse - The full API response.
+ *
+ * @example
+ * var name = 'new-subnetwork-name';
+ *
+ * var config = {
+ *   network : 'global/networks/network-name',
+ *   ipCidrRange: '10.0.1.0/24'
+ * };
+ *
+ * region.createSubnetwork(name, config,
+ *   function (err, subnetwork, operation, apiResponse) {
+ *   // `subnetwork` is a Subnetwork object.
+ *
+ *   // `operation` is an Operation object that can be used to check the status
+ *   // of the request.
+ * });
+ */
+Region.prototype.createSubnetwork = function(name, config, callback) {
+  var self = this;
+
+  var body = extend({}, config, {
+    name: name
+  });
+
+  this.request({
```

Comment:
Creating it through network messes up the url when using this.request which needs to be '/regions/{region}/subnetworks'.

I thought it made more sense to make it through the region object for this reason. However, it could be changed to be created through the Network object, i think it would just make the this.request have to be initiated in a strange way.